src/constants/MarkdownFiles/posts/gsoc-25-mostlyk-week01.md

```diff
@@ -0,0 +1,158 @@
+image: "assets/Images/GSOC.png"
+---
+
+
+<!-- markdownlint-disable -->
+
+# Community Bonding Period Progress Report by Krish Pandya
+
+**Project:** [GTK4 Exploration](https://summerofcode.withgoogle.com/programs/2025/projects/rsHsYZKy)  
+**Mentors:** [Walter Bender](https://github.com/walterbender) , [Ibiam Chihurumnaya Ibiam](https://github.com/chimosky)       
+**Organization:** Sugar Labs  
+**Reporting Period:** May 15, 2025 till June 4, 2025  
+
+---
+
+## Introduction & About Me
+
+Hey everyone! This is probably going to be my biggest blog post of the GSoC journey haha — and I'm excited to share my thoughts on the GTK4 migration project during the community bonding period and some work done till the first week. I will just give a short introduction—
+
+I'm Krish Pandya aka MostlyK, a B.Tech student in Electronics and Communication Engineering at IIITH. While my degree says ECE, I've found myself gravitating toward robotics, graphics, software , and most imporatant of them all open source development. When I'm not debugging ~~C++~~ (Rust/Python now) code or setting up new linux systems, you'll find me reading philosophy (currently juggling reading Sartre's Nausea and Red Rising), creating music, or tinkering with RISC-V processors. I believe in doing things properly , whether it's writing neat code, understanding the philosophy behind API changes, or setting up development workflows that actually make sense. And writing stuff so coming back , anyone can just give a glance and thank me for writing this, I know my future self is going to read this a lot!
+
+The Sugar desktop environment currently relies on GTK3, which is approaching end-of-life. My mission? Modernize Sugar by completing its migration to GTK4, ensuring long-term sustainability, improved performance, and access to modern UI features. This involves porting the core toolkit (sugar-toolkit-gtk3), porting an activity to GTK4 and making a new activity along with the new features that will be added and testing the changes.
+
+---
+
+## My Approach & Tooling Setup
+
+Before diving into the actual porting work, I spent considerable time setting up an efficient development workflow. Coming from other projects, I knew that having quick way to test my changes, that is currently I just have to build the toolkit first but later on testing , it helps to have a faster way to test stuff, I may even bring out some tmux scripts later down the weeks but as of now the following is enough:
+
+### Development Scripts That Saved My Sanity
+
+I created a couple of shell snippets that became essential during this period:
+
+```bash
+# Replace system sugar3 with development version
+# This ensures activities use the modified toolkit instead of the old system version
+sudo rm -rf /usr/lib/python3.13/site-packages/sugar3;sudo cp -r ./src/sugar3 /usr/lib/python3.13/site-packages/sugar3
+
+# Complete rebuild and install cycle
+sudo ./autogen.sh;sudo make clean;sudo make;sudo make install
```

Comment:
The copying is usually done after the build, I'm wondering why you're doing it before in this case.

Reply:
I can see that they're different commands, there's no mention of you clubbing them - what does clubbing mean?.

I'm fine with your blog report detailing your workflow, we won't need to change any of our docs to fit this when the time comes so it's cool.